### PR TITLE
[15.0][MIG] account_check_printing: migration script

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Supersedes https://github.com/OCA/OpenUpgrade/pull/3304.

### CODE REVIEW

File: openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from openupgradelib import openupgrade
+
+
+def _create_account_payment_method_line(env):
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Can you please express the logic behind adding these lines?

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: The key is in the line `WHERE apm.code = 'check_printing'`. The payment methods of this code need corresponding payment method lines.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: That's still no reason. Please indicate the code where that lines are needed, and why only for checks.

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: In `account` module, we have created the ones for:
Selection_722
In `payment` module, we have created the ones for:
Selection_721
So it's logical we need to populate in this PR the ones for `code = 'check_printing'`.
About why we need to create payment method lines for the payment methods... well, it's just the new methodology, we need records in this new model in order to make all this the payment method logic be functional. I mean, the payment methods in this account_check_printing module uses this code https://github.com/odoo/odoo/blob/14.0/addons/account_check_printing/models/account_journal.py#L69.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: OK, that's what the reference I was looking for. Please include it as comment in the code and then I'll merge it.

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Done

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: I don't see the code link in the comment...

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Added code link. The problem is that it was too long for flake8 tastes, but well, I put a shortened version.

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: It doesn't serve, as the line code is a changing one. Put the link in 2 lines or put the ignore directive.

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: What do you mean about "`(...)/account_check_printing/models/account_journal.py#L69`" doesn't serve? I don't understand you. It's super-pretty-obvious that `(...)` means "`https://github.com/odoo/odoo/tree/15.0/addons/`".

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: The line 69 today will be line XX on 2 months. You need an absolute link with the SHA hash in the link (the called permalink).

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Ah, and how I obtain that permalink?

Comment (pedrobaeza) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Selección_014

Comment (MiquelRForgeFlow) [reply openupgrade_scripts/scripts/account_check_printing/15.0.1.0/post-migration.py:4]: Thanks. Ready!

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**